lib/Dialect/TritonGPU/IR/Dialect.cpp

```diff
@@ -1979,18 +1983,25 @@ SmallVector<int64_t> NvidiaMmaEncodingAttr::getMMAv2Rep(ArrayRef<int64_t> shape,
                                                     warpsPerCTA[rank - 1]))};
   }
 }
+
 unsigned NvidiaMmaEncodingAttr::getTotalElemsPerThreadForOperands(
     ArrayRef<int64_t> shape, Type eltTy, int kWidth, int opIdx) const {
   auto shapePerCTA = getShapePerCTA(*this, shape);
   int warpsPerCTAM = getWarpsPerCTA()[0];
   int warpsPerCTAN = getWarpsPerCTA()[1];
   // H100
   if (isHopper()) {
-    return getTotalElemsPerThread(shape, eltTy);
+    assert(opIdx == 0);
```

Comment:
I earlier confirmed that the function `getTotalElemsPerThreadForOperands` is only invoked here:
```cpp
unsigned DotOperandEncodingAttr::getTotalElemsPerThread(ArrayRef<int64_t> shape,
                                                        Type eltTy) const {
  if (auto mmaParent = mlir::dyn_cast<MmaEncodingTrait>(getParent())) {
    return mmaParent.getTotalElemsPerThreadForOperands(shape, eltTy,
                                                       getKWidth(), getOpIdx());
```
And since I had a check in `DotOperandEncodingAttr`:
```cpp
    if (opIdx != 0 && parentAttr.isHopper())
      return emitError()
```
There cannot be a case where a dotOp has `opIdx==1` and a Hopper parent, and so the `opIdx==1` code flow cannot happen in the above function.

IMO it's appropriate to assert here since `getTotalElemsPerThreadForOperands()` calculates the number of elements stored by each thread in registers, which is logically inconsistent for Hopper WGMMA's second operand (opIdx 1), as it is always in shared memory.

Reply:
Sounds good thanks for checking!